src/ts/core/features/filter-search.ts

```diff
@@ -0,0 +1,130 @@
+import {Feature, Settings} from '../settings'
+import {browser} from 'webextension-polyfill-ts'
+import {waitForSelectorToExist} from 'src/core/common/mutation-observer'
+
+export const config: Feature = {
+    id: 'filter_search',
+    name: 'Search pages in page or references filter',
+    enabledByDefault: true,
+}
+
+const checkSettingsAndSetupFilterSearchToggle = () => {
+    Settings.isActive('filter_search').then(active => {
+        toggleFilterSearch(active)
+    })
+}
+checkSettingsAndSetupFilterSearchToggle()
+
+browser.runtime.onMessage.addListener(async message => {
+    if (message === 'settings-updated') {
+        checkSettingsAndSetupFilterSearchToggle()
+    }
+})
+
+const toggleFilterSearch = (active: boolean) => {
+    if (active) {
+        document.addEventListener('click', filterSearch)
+    } else {
+        document.removeEventListener('click', filterSearch)
+    }
+}
+
+const searchInputId = 'searchInputBox'
+
+const getSearchInput = (): HTMLInputElement | null => {
+    return document.getElementById(searchInputId) as HTMLInputElement
+}
+
+async function filterSearch(event: MouseEvent) {
+    const eventTarget = event.target as HTMLElement
+    if (!eventTarget) return
+
+    const filterPageButtonClicked = eventTarget.className === 'bp3-button'
```

Comment:
I think `src/ts/core/roam/selectors.ts` has `'bp3-button'`

Reply:
Hmm, is there a nice way to check if an element or one of it's children is matching a selector?